Headers/AppKit/NSPasteboardItem.h

```diff
@@ -0,0 +1,81 @@
+/** <title>NSPasteboardItem</title>
+
+   <abstract>class for wrapping pasteboard content</abstract>
+
+   Copyright <copy>(C) 2017 Free Software Foundation, Inc.</copy>
+
+   Author: Daniel Ferreira <dtf@stanford.edu>
+   Date: July 2017
+
+   This file is part of the GNUstep GUI Library.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Library General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Library General Public
+   License along with this library; see the file COPYING.LIB.
+   If not, write to the Free Software Foundation,
+   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#ifndef _GNUstep_H_NSPasteboardItem
+#define _GNUstep_H_NSPasteboardItem 
+
+#import <AppKit/AppKitDefines.h>
+#import <AppKit/NSPasteboard.h>
+#import <Foundation/NSArray.h>
+#import <Foundation/NSDictionary.h>
+#import <CoreFoundation/CFBase.h>
+
+#if OS_API_VERSION(MAC_OS_X_VERSION_10_6, GS_API_LATEST)
+
+@protocol NSPasteboardItemDataProvider;
+
+@interface NSPasteboardItem : NSObject <NSPasteboardWriting, NSPasteboardReading> {
+  NSMutableDictionary *_providerMap;
+  NSMutableDictionary *_dataMap;
+  NSMutableArray *_types;
+}
+#if GS_HAS_DECLARED_PROPERTIES
+@property (readonly, copy) NSArray *types;
```

Comment:
Alright. I did some reading (I am on the phone, so I cannot try this), and I applied some thinking:

- relevant section in [document about encapsulation](https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/EncapsulatingData/EncapsulatingData.html#//apple_ref/doc/uid/TP40011210-CH5-SW5) talks about setters changing behaviour but unless I missed something, not getters; I don't see evidence we getters return a copy (other than what was copied in setter)
- getters should be cheap, and making a new object every time a getter is called seems comparatively expensive

I think "readonly, copy" is identical to "readonly, assign" unless you redeclare the property as "copy" inside implementation. And it makes a difference if you use synthesize.

Either way, let's leave it as "readonly, copy" if the documented api says it should be so.